src/forces.f90

```diff
@@ -13,522 +13,832 @@
+             stressx = stressx + fdix
+             convy = convy + fcvy
+             stressy = stressy + fdiy
+             pressy = pressy + fpry
+          endif
+ 
+          !calculating the flux through the front wall (z-direction)
+          if((kcvfr(iv).ge.xstart(3).and.(kcvfr(iv).le.xend(3)))) then
+             k = kcvfr(iv)-xstart(3)+1
+             fcvx=zero
+             fcvy=zero
+             fpry=zero
+             fdix=zero
+             fdiy=zero
+             do j=jcvlw_lx(iv), jcvup_lx(iv)
+                do i=icvlf_lx(iv), icvrt_lx(iv)-1
+                   !momentum flux
+                   uxmid = half*(ux1(i,j,k)+ux1(i+1,j,k))
+                   uymid = half*(uy1(i,j,k)+uy1(i+1,j,k))
+                   uzmid = half*(uy1(i,j,k)+uy1(i+1,j,k))
+                   fcvx= fcvx + uxmid*uzmid*dx*del_y(j)
+                   fcvy= fcvy + uymid*uzmid*dx*del_y(j)
+ 
+                   !viscous flux
+                   dudzmid = half*(tg1(i,j,k)+tg1(i+1,j,k))
+                   dwdxmid = half*(tc1(i,j,k)+tc1(i+1,j,k))
+                   dwdymid = half*(tf1(i,j,k)+tf1(i+1,j,k))
+                   dvdzmid = half*(th1(i,j,k)+th1(i+1,j,k))
+                   fdix = fdix + (xnu*(dudzmid+dwdxmid)*dx*del_y(j))
+                   fdiy = fdiy + (xnu*(dvdzmid+dwdymid)*dx*del_y(j))
+                enddo
+             enddo   
+             convx = convx + fcvx
+             stressx = stressx + fdix
+             convy = convy + fcvy
+             stressy = stressy + fdiy
+          endif
+ 
+          !calculating the flux through the front wall (z-direction)
+          if((kcvbk(iv).ge.xstart(3).and.(kcvbk(iv).le.xend(3)))) then
+             k = kcvbk(iv)-xstart(3)+1
+             fcvx=zero
+             fcvy=zero
+             fpry=zero
+             fdix=zero
+             fdiy=zero
+             do j=jcvlw_lx(iv), jcvup_lx(iv)
+                do i=icvlf_lx(iv), icvrt_lx(iv)-1
+                   !momentum flux
+                   uxmid = half*(ux1(i,j,k)+ux1(i+1,j,k))
+                   uymid = half*(uy1(i,j,k)+uy1(i+1,j,k))
+                   uzmid = half*(uy1(i,j,k)+uy1(i+1,j,k))
+                   fcvx= fcvx - uxmid*uzmid*dx*del_y(j)
+                   fcvy= fcvy - uymid*uzmid*dx*del_y(j)
+ 
+                   !viscous flux
+                   dudzmid = half*(tg1(i,j,k)+tg1(i+1,j,k))
+                   dwdxmid = half*(tc1(i,j,k)+tc1(i+1,j,k))
+                   dwdymid = half*(tf1(i,j,k)+tf1(i+1,j,k))
+                   dvdzmid = half*(th1(i,j,k)+th1(i+1,j,k))
+                   fdix = fdix - (xnu*(dudzmid+dwdxmid)*dx*del_y(j))
+                   fdiy = fdiy - (xnu*(dvdzmid+dwdymid)*dx*del_y(j))
+                enddo
+             enddo   
+             convx = convx + fcvx
+             stressx = stressx + fdix
+             convy = convy + fcvy
+             stressy = stressy + fdiy
+          endif
+ 
+          call MPI_ALLREDUCE(convx, convxt, 1, real_type, MPI_SUM, MPI_COMM_WORLD, code)
```

Comment:
Use MPI_IN_PLACE and check the error code

Reply:
Can you elaborate where to use MPI_IN_PLACE and what error code should I check for ?

Reply:
The variable `convxt` is not needed if the MPI reduction is performed in-place.
https://www.mpi-forum.org/docs/mpi-2.2/mpi22-report/node109.htm
https://github.com/xcompact3d/Incompact3d/blob/1cfae4edd9cd36d1b36934dd3398d6f6d1cb0c86/src/navier.f90#L1511

I would recommend to check the MPI error code after calling a MPI function. Currently, most of the MPI error codes are not checked in xcompact3d.

Reply:
Got it ! Thanks ! Will make these changes and create a new PR.